synapse/server.py

```diff
@@ -299,8 +300,11 @@ def is_mine(self, domain_specific_string: DomainSpecificString) -> bool:
     def is_mine_id(self, string: str) -> bool:
         return string.split(":", 1)[1] == self.hostname
 
+    # The caching attribute for this is technically already set in __init__, but it's replicated for the sake of
+    # consistency
+    @cache_in_self
     def get_clock(self) -> Clock:
-        return self.clock
+        return Clock(self._reactor)
```

Comment:
That's actually sorta-not-really possible, as line 241 (`clock=self._clock,` up there) makes it so this function has to be called from within `__init__`, and i'm kinda queasy to call a `@cache_in_self`-cached function from within `__init__`

Reply:
> i'm kinda queasy to call a @cache_in_self-cached function from within __init__

ok, but why? it seems no different to calling any other function from within __init__ to me.

failing that: have this method raise a NotImplementedError or something, instead of having code that isn't used. But I think you should just call `get_clock` from inside __init__.

Reply:
I'm queasy to call it, because it's effectively (in my mind) operating some cached function/attribute tricks on, what is effectively, a not-ready object, an object that is not finalised in its initialisation. I can't know if its okay to use `get_clock` in `__init__` because i dont know how `@cache_in_self` will operate if the `self` it is operating on is *sorta* not finalised?

Basically: I need you or someone else's assurance that it'll go alright, cuz i can't know for sure rn.

Reply:
> I'm queasy to call it, because it's effectively (in my mind) operating some cached function/attribute tricks on, what is effectively, a not-ready object, an object that is not finalised in its initialisation. I can't know if its okay to use `get_clock` in `__init__` because i dont know how `@cache_in_self` will operate if the `self` it is operating on is _sorta_ not finalised?

well, these things are all true whenever you call a method on `self` from within `__init__`, and we don't make any attempt to forbid such a practice. We just say "you need to be careful, but if it makes the code cleaner then it's ok", and the same applies here.

> Basically: I need your or someone else's assurance that it'll go alright, cuz i can't know for sure rn.

we can only tell that by inspecting `cache_in_self` (and ideally testing to confirm). You're modifying `cache_in_self` so you ought to be capable of figuring it out for yourself. I don't understand why you need me to tell you.

Reply:
hmmm, ill take another closer look at it later then